doc/raspbian_install.md

```diff
@@ -0,0 +1,130 @@
+
+We are going to install the app into `/etc/goblin` to isolate it from the rest of the system. (It'll be served over nginx when you do things for real.)
+
+```sh
+sudo adduser goblin
+sudo bash
+cd /etc
+sudo git clone https://github.com/johnHackworth/goblin.git
+sudo chown -R goblin:goblin goblin
+```
+
+## Getting pnpm running
+
+The first thing you'll need to do is get `corepack` up and running so that it can get `pnpm` up and running. We'll install `corepack` underneath `/etc/goblin`.
+
+Note that `pnpm` also needs to do some `npm` stuff, and you'll need to be able to write to a `.npm` directory in $HOME, so let's get that out of the way first:
+
+```sh
+sudo -u goblin bash
+cd /etc/goblin
+corepack use pnpm@8
+```
+
+If you are trying to install to a directory that's on an SD card, go get several cups of coffee and a fun YouTube playlist. You'll be here a while waiting for `corepack use` to finish.
+
+...and then you will find the install fails halfway through! Cypress thinks that you don't satisfy its system requirements. (The Cypress devs do not think people want to use it on a Pi, so their install script doesn't recognize the `armv7` architecture. Their install script defaults to assuming any architecture it doesn't recognize is ia32.) You must fool it. Go into `node_modules/cypress/lib/util.js` and edit the `getRealArch()` function:
+
+```javascript
+async getRealArch() {
+    return 'arm64';
+    // and then the rest of it
+}
+```
+
+Cypress will download a binary that isn't compatible with your Pi, which means you won't be able to run any tests. But you'll at least have succeeded installing pnpm.
+
+## Installing Rust
+
+The `packages/backend/native-utils/` directory contains, well, some native code. It needs `cargo` to be installed. The `cargo` from Raspbian isn't a recent enough version to build with, so you're going to need to install it via the Rust website. We'll install it just for the `goblin` user rather than system-wide for the moment.
+
+```sh
+sudo -u goblin bash
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+
+## Building goblin
+
+Time to build!
+
+```sh
+sudo -u goblin bash
+cd /etc/goblin
+source /home/goblin/.cargo/env
+pnpm run build
+```
+
+## Databases
+
+Now let's get the database set up. We're going to go with `goblin` for everything (database name, username, password).
+
+**DO NOT DO THIS IN PRODUCTION**. Even if you _think_ you know your Postgres is nailed down to prevent outside connections. Hope you enjoy your Postgres install housing the metadata for North Korean hacking teams.
+
+```sql
+CREATE USER goblin WITH PASSWORD 'goblin';
+CREATE DATABASE goblin WITH encoding = 'UTF8';
+GRANT ALL ON DATABASE goblin TO goblin;
+```
+
+Now run the migrate step to create the tables (and there's a lot of them):
+
+```sh
+sudo -u goblin bash
+cd /etc/goblin
+pnpm run migrate
```

Comment:
doesn't it expect `pnpm run init` first, or the init only runs `migrate`? Doesn't matter probably, and i'm too lazy to check now

Reply:
Gonna put it in for the moment, and then maybe later I'll try these steps from a stock Raspbian image to test it.